cli/base_commands.py

```diff
@@ -118,3 +124,44 @@ def get_help_string(self) -> str:
     async def execute_default(self, args, context):
         print('Invalid usage. Provide a subcommand.')
         return CommandResultNone()
+
+
+class ScanCommand(Command):
+    def get_help_string(self) -> str:
+        return 'Perform scan for TCAT devices'
+
+    async def execute_default(self, args, context):
+        if not (context['ble_sstream'] is None):
+            del context['ble_sstream']
+
+        tcat_devices = await ble_scanner.scan_tcat_devices()
+        if tcat_devices:
+            print('Found devices:\n')
+            for i, device in enumerate(tcat_devices):
+                print(f'{i + 1}: {device.name} - {device.address}')
+        else:
+            print('\nNo devices found.')
+            return CommandResultNone()
+
+        print('\nSelect the target number to connect to it.')
+        selected = get_int_in_range(1, len(tcat_devices))
+        device = tcat_devices[selected - 1]
+        print('Selected ', device)
```

Comment:
Suggestion: maybe this could be moved into `scan_tcat_devices`, returning `None` instead of `CommandResultNone` and device if successful. This way we could just call 
```
device = await ble_scanner.scan_tcat_devices()
if device is None:
    return CommandResultNone()
```
here, and here: https://github.com/pjasinski990/bbtc/blob/main/bbtc.py#L96 just 
```
device = await ble_scanner.scan_tcat_devices()
```
if `None` will be returned here, program will return back to CLI instead of quitting (as it does for searching by name or MAC)